ros2pkg/ros2pkg/build_type/ament_cmake.py

```diff
@@ -0,0 +1,129 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+from ros2pkg.api import PackageCreationError
+from ros2pkg.api.build_type import BaseBuildType
+from ros2pkg.api.create import create_folder
+from ros2pkg.api.create import create_file
+
+
+class AmentCMakeBuildType(BaseBuildType):
+    """
+    Create a ament_cmake package.
+
+    Subclasses should override _basic_populate() with custom
+    package directory population logic.
+
+    :ivar include_directory: The package directory for C include files.
+    """
+
+    def __init__(self):
+        super(AmentCMakeBuildType, self).__init__()
+        self.include_directory = None
+
+    def init(self, args):
+        super(AmentCMakeBuildType, self).init(args)
+
+        if self.args.library_name:
+            self.buildtool_depends = ['ament_cmake_ros']
+        self.test_dependencies = ['ament_lint_auto', 'ament_lint_common']
+
+    def populate(self):
+        # call super() for src and include directory creation
+        super(AmentCMakeBuildType, self).populate()
+
+        # create the primary package artifacts
+        self._basic_populate()
+
+        if self.args.node_name:
+            self._populate_cpp_node()
+
+        if self.args.library_name:
+            self._populate_cpp_library()
+
+    def create_source_folders(self):
+        super(AmentCMakeBuildType, self).create_source_folders()
+
+        print('creating source and include folder')
+        self.source_directory = create_folder('src', self.package_directory)
+        if not self.source_directory:
+            return f'unable to create source folder in {self.package_directory}'
+
+        self.include_directory = \
+            create_folder(self.package.name, self.package_directory + os.sep + 'include')
+        if not self.include_directory:
+            raise PackageCreationError(
+                    f'unable to create source or include folder in {self.package_directory}')
+
+    def _basic_populate(self):
+        return self._populate_ament_cmake()
+
+    def _populate_ament_cmake(self):
+        cmakelists_config = {
+            'project_name': self.package.name,
+            'dependencies': [str(dep) for dep in self.package.build_depends],
+            'cpp_node_name': self.args.node_name,
+            'cpp_library_name': self.args.library_name,
+        }
+        create_file('ros2pkg',
+                    'ament_cmake/CMakeLists.txt.em',
+                    self.package_directory,
+                    'CMakeLists.txt',
+                    cmakelists_config)
```

Comment:
I think we prefer hanging indentation as per [PEP 8 style guide](https://www.python.org/dev/peps/pep-0008/#indentation) 
```
create_file(
    'ros2pkg',
    'ament_cmake/CMakeLists.txt.em',
    self.package_directory,
    'CMakeLists.txt',
    cmakelists_config)
```
I did notice some code from master didn't strictly follow the style guide so sorry about the confusion.

Reply:
updated project code for PEP 8 consistency.